plugins/apitracing/src/lib/ApiTracing.cpp

```diff
@@ -0,0 +1,121 @@
+#include "Config.h"
+#include "Filenames.h"
+#include "Tracer.h"
+#include "config/TracingTargetsParser.h"
+#include "os/windows/Library.h"
+#include <fmt/core.h>
+#include <tclap/CmdLine.h>
+#include <vmicore/plugins/PluginInit.h>
+#include <vmicore/plugins/PluginInterface.h>
+
+VMI_PLUGIN_API_VERSION_INFO
+
+namespace ApiTracing
+{
+    namespace
+    {
+        VmiCore::Plugin::PluginInterface* pluginInterface;
+        std::shared_ptr<Tracer> tracer;
+    }
+
+    void processStartCallback(std::shared_ptr<const VmiCore::ActiveProcessInformation> processInformation)
+    {
+        pluginInterface->logMessage(
+            VmiCore::Plugin::LogLevel::info,
+            LOG_FILENAME,
+            fmt::format("Starting process: {} with pid: {}", *processInformation->fullName, processInformation->pid));
+        tracer->addHooks(*processInformation);
+    }
+
+    void shutDownCallback()
+    {
+        tracer->removeHooks();
+    }
+
+    extern "C" bool
+    init(VmiCore::Plugin::PluginInterface* communicator,
+         std::shared_ptr<VmiCore::Plugin::IPluginConfig> config, // NOLINT(performance-unnecessary-value-param)
+         std::vector<std::string> args)
+    {
+        pluginInterface = communicator;
+
+        TCLAP::CmdLine cmd("ApiTracing plugin for VMICore.", ' ', PLUGIN_VERSION);
```

Comment:
It makes sense to add a default value here in case `PLUGIN_VERSION` is undefined. Right now this evaluates to an empty string if I'm not mistaken. https://github.com/GDATASoftwareAG/smartvmi/blob/main/plugins/inmemoryscanner/src/lib/InMemory.cpp#L1